src/proto/types.rs

```diff
@@ -72,44 +72,83 @@ pub struct CloseCode(NonZeroU16);
 impl CloseCode {
     /// Normal closure, meaning that the purpose for which the connection was
     /// established has been fulfilled.
-    pub const NORMAL_CLOSURE: Self = Self(unsafe { NonZeroU16::new_unchecked(1000) });
+    pub const NORMAL_CLOSURE: Self = Self(match NonZeroU16::new(1000) {
+        Some(val) => val,
+        None => unreachable!()
+    });
```

Comment:
Is it not better to either just do `Self(NonZeroU16::new(1000).unwrap());` or to add a private `new` function for `CloseCode`?

Reply:
A `new_internal` style function would also work :+1: